src/main/java/frc/robot/subsystems/arm/ArmIOReal.java

```diff
@@ -123,235 +144,205 @@ public void setShoulderJointPower(double power) {
+                ySetpoint = armLocation.getY();
+            }
+
+            /**
+             * Gets the velocity of the shoulder motors.
+             *
+             * @return shoulder motors velocity. [rad/sec]
+             */
+            public double getShoulderMotorVelocity () {
+                return shoulderMainMotor.getVelocity().getValue();
+            }
+
+            public void setFinalSetpointAngles (Translation2d position, ArmInputsLogged inputs){
+                var solution = kinematics.inverseKinematics(position);
+                inputs.finalSetpointAngles[0] = Math.toDegrees(solution.shoulderAngle);
+                inputs.finalSetpointAngles[1] = Math.toDegrees(solution.elbowAngle);
+            }
+
+            /**
+             * Gets the velocity of the elbow motors.
+             *
+             * @return elbow motor velocity. [rad/sec]
+             */
+            public double getElbowMotorVelocity () {
+                return elbowMainMotor.getVelocity().getValue();
+            }
+
+            public void resetArmEncoders () {
+                double elbowAngleReset = elbowEncoder.getAbsolutePosition() * 360.0 - ArmConstants.ELBOW_ZERO_POSITION;
+                double shoulderAngleReset = shoulderEncoder.getAbsolutePosition() * 360.0 - ArmConstants.SHOULDER_ZERO_POSITION;
+                shoulderOffset = AngleUtil.normalize(shoulderAngleReset) / 360.0;
+                elbowOffset = AngleUtil.normalize(elbowAngleReset) / 360.0;
+            }
+
+            public Translation2d getElbowJointPosition () {
+                Rotation2d shoulderAngle = getShoulderJointAngle();
+                return new Translation2d(
+                        ArmConstants.SHOULDER_ARM_LENGTH * shoulderAngle.getCos(),
+                        ArmConstants.SHOULDER_ARM_LENGTH * shoulderAngle.getSin());
+            }
+
+            public boolean armIsOutOfFrame () {
+                Translation2d elbowJoint = getElbowJointPosition(), endPosition = getEndPosition();
+                return !(elbowJoint.getX() < 0) || !(endPosition.getX() < 0);
+            }
+
+            public boolean armIsInRobot () {
+                return getEndPosition().getY() < 0 && getShoulderJointAngle().getDegrees() > 90;
+            }
+
+            public void setVelocity (Translation2d velocity){
+                var armVelocities = kinematics.getVelocities(getEndPosition(), velocity);
+                setShoulderJointPower(armVelocities.shoulderAngle / Math.PI);
+                setElbowJointPower(armVelocities.elbowAngle / (2 * Math.PI));
+            }
+
+            public boolean changedToDefaultCommand () {
+                return changedToDefaultCommand;
+            }
+
+            /**
+             * Stops the motors.
+             */
+            public void stop () {
+                shoulderMainMotor.set(0);
+                elbowMainMotor.set(0);
```

Comment:
[`TalonFX.stopMotor()`](https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/hardware/TalonFX.html#stopMotor()) is more idiomatic here.